File: cotsSource/org.apache.qpid/java/common/src/main/java/org/apache/qpid/transport/util/Logger.java
/*
 *
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 *
 */
package org.apache.qpid.transport.util;

import org.slf4j.LoggerFactory;

/**
 * Logger
 *
 */

public final class Logger
{

    public static final Logger get(Class<?> klass)
    {
        return new Logger(LoggerFactory.getLogger(klass));
    }

    private final org.slf4j.Logger log;

    private Logger(org.slf4j.Logger log)
    {
        this.log = log;
    }

    public boolean isDebugEnabled()
    {
        return log.isDebugEnabled();
    }

    public void debug(String message, Object ... args)
    {
        if (log.isDebugEnabled())
        {
            log.debug(String.format(message, args));
        }
    }

    public void debug(Throwable t, String message, Object ... args)
    {
        if (log.isDebugEnabled())
        {
            log.debug(String.format(message, args), t);
        }
    }

    public void error(String message, Object ... args)
    {
        if (log.isErrorEnabled())
        {
            log.error(String.format(message, args));
        }
    }

    public void error(Throwable t, String message, Object ... args)
    {
        if (log.isErrorEnabled())
        {
            log.error(String.format(message, args), t);
        }
    }

    public void warn(String message, Object ... args)
    {
        if (log.isWarnEnabled())
        {
            log.warn(String.format(message, args));
        }
    }

    public void warn(Throwable t, String message, Object ... args)
    {
        if (log.isWarnEnabled())
        {
            log.warn(String.format(message, args), t);
        }
    }

    public void info(String message, Object ... args)
    {
        if (log.isInfoEnabled())
        {
            log.info(String.format(message, args));
        }
    }

    public void info(Throwable t, String message, Object ... args)
    {
        if (log.isInfoEnabled())
        {
            log.info(String.format(message, args), t);
        }
    }

    public void trace(String message, Object ... args)
    {
        if (log.isTraceEnabled())
        {
            log.trace(String.format(message, args));
        }
    }

    public void trace(Throwable t, String message, Object ... args)
    {
        if (log.isTraceEnabled())
        {
            log.trace(String.format(message, args), t);
        }
    }

}

File: cotsSource/org.apache.qpid/dotnet/client-010/client/transport/util/Logger.cs
/*
*
* Licensed to the Apache Software Foundation (ASF) under one
* or more contributor license agreements.  See the NOTICE file
* distributed with this work for additional information
* regarding copyright ownership.  The ASF licenses this file
* to you under the Apache License, Version 2.0 (the
* "License"); you may not use this file except in compliance
* with the License.  You may obtain a copy of the License at
*
*   http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing,
* software distributed under the License is distributed on an
* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
* KIND, either express or implied.  See the License for the
* specific language governing permissions and limitations
* under the License.
*
*/
using System;
using log4net;

namespace org.apache.qpid.transport.util
{

    /// <summary> Logger
    /// 
    /// </summary>

    public sealed class Logger
    {
        private readonly ILog log;

        public static Logger Get(Type type)
        {
            return new Logger(LogManager.GetLogger(type));
        }

        private Logger(ILog log)
        {
            this.log = log;
        }

        public bool IsDebugEnabled()
        {
            return log.IsDebugEnabled;
        }

        public void Debug(String message, params Object[] args)
        {
            if (log.IsDebugEnabled)
            {
                log.Debug(String.Format(message, args));
            }
        }

        public void Debug(Exception t, String message, params Object[] args)
        {
            if (log.IsDebugEnabled)
            {
                log.Debug(String.Format(message, args), t);
            }
        }

        public void Error(String message, params Object[] args)
        {
            if (log.IsErrorEnabled)
            {
                log.Error(String.Format(message, args));
            }
        }

        public void Error(Exception t, String message, params Object[] args)
        {
            if (log.IsErrorEnabled)
            {
                log.Error(String.Format(message, args), t);
            }
        }

        public void Warn(String message, params Object[] args)
        {
            if (log.IsWarnEnabled)
            {
                log.Warn(String.Format(message, args));
            }
        }

        public void Warn(Exception t, String message, params Object[] args)
        {
            if (log.IsWarnEnabled)
            {
                log.Warn(String.Format(message, args), t);
            }
        }

        public void Info(String message, params Object[] args)
        {
            if (log.IsInfoEnabled)
            {
                log.Info(String.Format(message, args));
            }
        }

        public void Info(Exception t, String message, params Object[] args)
        {
            if (log.IsInfoEnabled)
            {
                log.Info(String.Format(message, args), t);
            }
        }
    }
}
